main.js

```diff
@@ -1,58 +1,185 @@
-const palindrome = (word) => {
+const daysInMonth = 30;
+const dayOfWeek = 2;
+const getCalendarMonth = (daysInMonth, dayOfWeek) => {
+  let day = daysInMonth - dayOfWeek + 1;
+  let dates = [];
+  for (let i = 0; i < 5; i++) {
+    const week = [];
+    for (let j = 0; j < 7; j++) {
+      if (dayOfWeek > daysInMonth || dayOfWeek > 7) {
+        return 'Error';
+      } else if (day > 30) {
+        day = 1;
+      }
+      week.push(day);
+      day++;
+    }
+    dates.push(week);
+  }
+  return dates;
+};
+
+const calendarMonth = getCalendarMonth(daysInMonth, dayOfWeek);
+let daysArray = calendarMonth.flat();
+
+let selectedDate = (arr, checkInDate, checkOutDate) => {
+  let resDate = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (checkInDate === i) {
+      resDate.push(true);
+    } else if (checkInDate > 31) {
+      return `Error day`;
+    } else if (checkOutDate === i && checkOutDate > checkInDate) {
+      resDate.push(true);
+    } else if (checkOutDate < checkInDate) {
+      return `Error date`;
+    } else resDate.push(false);
+  }
+  return resDate;
+};
+
+let selectedDateRes = selectedDate(daysArray, 3, 5);
+const days = [{}];
+const selected = [{}];
+daysArray.forEach((item) => days.push({ dayOfMonth: item }));
+selectedDateRes.forEach((item) => selected.push({ selectedDay: item }));
+
+const calendar = days.map((item, index) => ({
+  ...item,
+  ...selected[index],
+}));
+console.log(calendar);
+
+console.log('----LESSON-8----');
+const studentsData = [
+  {
+    firstName: 'Василий',
+    lastName: 'Петров',
+    admissionYear: 2019,
+    courseName: 'Java',
+  },
+  {
+    firstName: 'Иван',
+    lastName: 'Иванов',
+    admissionYear: 2018,
+    courseName: 'JavaScript',
+  },
+  {
+    firstName: 'Александр',
+    lastName: 'Федоров',
+    admissionYear: 2017,
+    courseName: 'Python',
+  },
+  {
+    firstName: 'Николай',
+    lastName: 'Петров',
+    admissionYear: 2019,
+    courseName: 'Android',
+  },
+];
+
+class User {
+  constructor(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
+
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+
+const currentYear = new Date().getFullYear();
+
+class Student extends User {
+  constructor(firstName, lastName, admissionYear, courseName) {
+    super(firstName, lastName);
+    this.admissionYear = admissionYear;
+    this.courseName = courseName;
+  }
+
+  get course() {
+    return currentYear - this.admissionYear;
+  }
+}
 
-    let long = word.length;
-    for (let i = 0; i < (long / 2); i++) {
-        console.log(i);
-        if (word[i] !== word[long - 1 - i]) {
-            return false;
-        }
+const studentList = studentsData.map(
+  ({ firstName, lastName, admissionYear, courseName }) =>
+    new Student(firstName, lastName, admissionYear, courseName),
+);
```

Comment:
Look at the example of the task. We should create the object Students with studentsData directly. You may put this logic inside the class.